AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -193,9 +201,20 @@ - (void)propagateUpLayoutable:(id<ASLayoutable>)layoutable
   }
 }
 
+- (ASEnvironmentTraitCollection)environmentTraitCollection
+{
+  return _environmentState.traitCollection;
```

Comment:
I think the locking may be unsafe in this file, BUT - if we do lock for accessing _environmentState below in asyncTraitCollection, we probably should lock here